util/flocks/src/lib.rs

```diff
@@ -1,2 +1,37 @@
 pub mod frwlock;
-pub mod tfrwlock;
\ No newline at end of file
+pub mod tfrwlock;
+
+pub mod asynchronous {
+
+    use rustix::{
+        fd::AsFd,
+        fs::{flock as sync_flock, FlockOperation}
+    };
+    use thiserror::Error;
+
+    #[derive(Debug, Error)]
+    pub enum AsyncFlockError {
+        #[error("Lock is not available")]
+        JoinError(#[from] tokio::task::JoinError),
```

Comment:
Future-proofing: this might be leaking too much implementation detail. If we decide we don't need `spawn_blocking`, this variant will have to be changed. Maybe just an opaque `NotAvailable` variant with a message would do.